### PR TITLE
crimson/osd: cleanup Notify::send_completion()

### PR DESCRIPTION
**Pull request by tchaikov**
## Checklist
- [ ] References tracker ticket
- [ ] Updates documentation if necessary
- [ ] Includes tests for new functionality or reproducer for bug

---

<details>
<summary>Show available Jenkins commands</summary>

- `jenkins retest this please`
- `jenkins test classic perf`
- `jenkins test crimson perf`
- `jenkins test signed`
- `jenkins test make check`
- `jenkins test make check arm64`
- `jenkins test submodules`
- `jenkins test dashboard`
- `jenkins test api`
- `jenkins test docs`
- `jenkins render docs`
- `jenkins test ceph-volume all`
- `jenkins test ceph-volume tox`

</details>

### CODE REVIEW

**Comment by tchaikov** [comment]
BTW, i realized `Watch::discard()` is still missing, which is called when interval changes. see `context_registry_on_change()`.

File: src/crimson/osd/watch.h [hunk 1]
```diff
@@ -202,14 +207,16 @@ seastar::future<> Notify::create_n_propagate(
...
                    crimson::osd::WatchRef>);
-  auto notify = seastar::make_shared<Notify>(
-    private_ctag_t{},
-    begin,
-    end,
-    std::forward<Args>(args)...);
   if (begin == end) {
+    auto notify = seastar::make_shared<Notify>(
+      private_ctag_t{},
+      std::forward<Args>(args)...);
```
Comment (rzarzynski) [inline src/crimson/osd/watch.h:213]: Makes sense.

File: src/crimson/osd/watch.cc [hunk 2]
```diff
@@ -173,8 +176,6 @@ seastar::future<> Notify::send_completion(
   logger().info("{} -- {} in progress watchers, {} timedout watchers {}",
                 __func__, watchers.size(), timedout_watchers.size());
   logger().debug("{} sending notify replies: {}", __func__, notify_replies);
-  complete = true;
-  timeout_timer.cancel();
```
Comment (rzarzynski) [inline src/crimson/osd/watch.cc]: :+1:

**Review by rzarzynski — APPROVED** [review]
LGTM. Thanks!

**Pull request merged**